src/eth.rs

```diff
@@ -462,6 +462,8 @@ where
                     spi_host_id: host,
                     spi_devcfg: &spi_devcfg as *const _ as *mut _,
                     int_gpio_num: int,
+                    #[cfg(all(esp_idf_version_major = "5", esp_idf_version_minor = "2",))]
```

Comment:
This is not OK, as we'll have to touch it again once 5.3 or 6.0 is out. Rather, replace it with `not(<the upper condition>)`.